pkg/controller/association/reconciler.go

```diff
@@ -209,11 +209,18 @@ func (r *Reconciler) Reconcile(ctx context.Context, request reconcile.Request) (
 	}
 
 	// we want to attempt a status update even in the presence of errors
-	if err := r.updateStatus(ctx, associated, newStatusMap); err != nil {
-		if apierrors.IsConflict(err) {
-			log.V(1).Info("Conflict while updating status")
-			return results.WithResult(reconcile.Result{Requeue: true}).Aggregate()
-		}
+	if err := r.updateStatus(ctx, associated, newStatusMap); err != nil && apierrors.IsConflict(err) {
+		log.V(1).Info(
+			"Conflict while updating status",
+			"namespace", associatedKey.Namespace,
+			"name", associatedKey.Name)
+		return results.WithResult(reconcile.Result{Requeue: true}).Aggregate()
+	} else if err != nil {
+		log.Error(
```

Comment:
I think this will lead to the error being logged twice. Once here and once through controller-runtime. I think it would be preferable to maybe wrap the error instead with some context information and stack trace e.g. through `pkg/errors` and pass it on to be logged on the top level.